src/jdk.jpackage/windows/classes/jdk/jpackage/internal/WixTool.java

```diff
@@ -32,105 +31,198 @@
 import java.nio.file.Path;
 import java.nio.file.PathMatcher;
 import java.text.MessageFormat;
-import java.util.Collections;
 import java.util.Comparator;
-import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Objects;
 import java.util.Optional;
-import java.util.function.Supplier;
+import java.util.Set;
+import java.util.function.Function;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
+import jdk.jpackage.internal.WixToolset.WixToolsetType;
 
 /**
  * WiX tool.
  */
 public enum WixTool {
-    Candle, Light;
+    Candle3("candle", DottedVersion.lazy("3.0")),
+    Light3("light", DottedVersion.lazy("3.0")),
+    Wix4("wix", DottedVersion.lazy("4.0.4"));
```

Comment:
Is there any reason to use "4.0.4" here? Latest released version is "4.0.5". Can we just put "4.0" similar to "3.0".